RecoTracker/MeasurementDet/plugins/MeasurementTrackerESProducer.cc

```diff
@@ -50,6 +51,12 @@ MeasurementTrackerESProducer::produce(const CkfComponentsRecord& iRecord)
   std::string stripCPEName = pset_.getParameter<std::string>("StripCPE");
   std::string matcherName  = pset_.getParameter<std::string>("HitMatcher");
 
+  //FIXME:: just temporary solution for phase2!
+  std::string phase2TrackerCPEName = "";
+  if (pset_.existsAs<std::string>("Phase2StripCPE")) {
+    phase2TrackerCPEName = pset_.getParameter<std::string>("Phase2StripCPE");
+  }
```

Comment:
I also prefer parsing `ParameterSet` in the constructor. But I want to ask clarification for "no need to re-get every event". Isn't framework supposed to cache ESProducts as long as they're valid? Naively I'd expect everything in `CkfComponentsRecord` to be valid for the entire job and thus constructed only once (so in practice there maybe isn't any performance gain?).